docs/docs/file-system-page-creation.md

```diff
@@ -0,0 +1,167 @@
+
+Developers can create multiple pages from a model based on the collection of nodes within it. To do that, use curly braces (`{ }`) to signify dynamic URL segments that relate to a field within the node. There is some special logic that can happen in here. Here are a few examples:
+
+- `/src/pages/products/{Product.name}.js => /products/burger`
+- `/src/pages/products/{Product.fields__sku}.js => /products/001923`
+- `/src/pages/blog/{MarkdownRemark.parent__(File)__relativePath}.js => /blog/learning-gatsby`
+
+We use the content within the curly braces to generate GraphQL queries to retrieve the nodes that should be built for a given collection. For example:
+
+`/src/pages/products/{Product.name}.js` generates the following query:
+
+```graphql
+allProduct {
+  nodes {
+    id # Gatsby always queries for id
+    name
+  }
+}
+```
+
+`/src/pages/products/{Product.fields__sku}.js` generates the following query:
+
+```graphql
+allProduct {
+  nodes {
+    id # Gatsby always queries for id
+    fields {
+      sku
+    }
+  }
+}
+```
+
+`/src/pages/blog/{MarkdownRemark.parent__(File)__relativePath}.js` generates the following query:
+
+```graphql
+allMarkdownRemark {
+  nodes {
+    id # Gatsby always queries for id
+    parent {
+      … on File {
+        relativePath
+      }
+    }
+  }
+}
+```
+
+This is the query that Gatsby uses to grab all the nodes and create a page for each of them. Gatsby also adds `id` to every query automatically to simplify how to integrate with page queries.
+
+## Component implementation
+
+Page components act the exact same way. Gatsby will create an instance of it for each node it finds in it’s querying. If a user needs to customize the query used for collecting the nodes, that can be done with a special export. Much akin to page queries. Here’s an example.
+
+```jsx:title=/src/pages/products/{Product.name}.js
+import { unstable_collectionGraphql } from "gatsby"
+
+export default function Component(props) {
+  return props.data.fields.sku + props.params.name
+}
+
+// If you are customizing the collection query, there is a special fragment you MUST use when using this API. The fragment converts to
+// { nodes { id, [params_from_path] } }
+export const collectionQuery = unstable_collectionGraphql`
+{
+  allProduct(filter: { type: { nin: ["Burger"] } }) {
+    ...CollectionPagesQueryFragment
+  }
+}`
+
+// This is the page query that connects the data to the actual component. Here you can query for any and all fields
+// you need access to within your code. Again, since Gatsby always queries for `id` in the collection, you can use that
+// to connect to this GraphQL query.
+export const query = graphql`
+  query ($id: String) {
+    product(id: { eq: $id }) {
+      fields {
+        sku
+      }
+    }
+  }
+}
+```
+
+## Routing and linking
+
+We "slugify" every route that is created from collection pages. When you want to link to one of those pages, it may not always be clear how to construct the URL from scratch.
```

Comment:
```suggestion
Gatsby "slugifies" every route that gets created from collection pages. When you want to link to one of those pages, it may not always be clear how to construct the URL from scratch.
```